internal/rename/connector.go

```diff
@@ -0,0 +1,59 @@
+/*
+ *  *******************************************************************************
+ *  * Copyright (c) 2019 Edgeworx, Inc.
+ *  *
+ *  * This program and the accompanying materials are made available under the
+ *  * terms of the Eclipse Public License v. 2.0 which is available at
+ *  * http://www.eclipse.org/legal/epl-2.0
+ *  *
+ *  * SPDX-License-Identifier: EPL-2.0
+ *  *******************************************************************************
+ *
+ */
+
+package rename
+
+import (
+	"fmt"
+	apps "github.com/eclipse-iofog/iofog-go-sdk/pkg/apps"
+	"github.com/eclipse-iofog/iofogctl/internal"
+	"github.com/eclipse-iofog/iofogctl/internal/config"
+	"github.com/eclipse-iofog/iofogctl/pkg/util"
+)
+
+type connectorExecutor struct {
+	namespace string
+	name      string
+	newname   string
+	filename  string
+}
+
+func newConnectorExecutor(namespace, name, newname string) *connectorExecutor {
+	c := &connectorExecutor{}
+	c.namespace = namespace
+	c.name = name
+	c.newname = newname
+	return c
+}
+
+func (exe *connectorExecutor) GetName() string {
+	return exe.name
+}
+
+func (exe *connectorExecutor) Execute() error {
+	// Check that Controller exists in current namespace
+	_, err := config.GetConnector(exe.namespace, exe.name)
+	if err != nil {
+		return err
+	}
+
+	util.SpinStart(fmt.Sprintf("Renaming Connector %s", exe.name))
+
+	// Do a shallow rename of controller
+	err = RenameConnector(exe.namespace, exe.name, exe.newname)
```

Comment:
Where is that function declared ? why is it not part of the executor ?

Reply:
In utils.go, I can move it into the executor function

Reply:
Yes, I'd rather have all code specific to one resource in its resource file.
Utils is used when the code is some sort of helper function